dali/python/nvidia/dali/_utils/eager_utils.py

```diff
@@ -547,26 +653,40 @@ def _wrap_stateless(op_class, op_name, wrapper_name):
     """
     def wrapper(*inputs, **kwargs):
         inputs, init_args, call_args = _prep_args(
-            inputs, kwargs, op_name, wrapper_name, _wrap_stateless.disqualified_arguments)
+            inputs, kwargs, op_name, wrapper_name, _callable_op_factory.disqualified_arguments)
 
         # Creating cache key consisting of operator name, description of inputs, input arguments
         # and init args. Each call arg is described by dtype, layout and dim.
         key = op_name + _desc_call_args(inputs, call_args) + str(sorted(init_args.items()))
 
         if key not in _stateless_operators_cache:
-            _stateless_operators_cache[key] = _stateless_op_factory(
+            _stateless_operators_cache[key] = _callable_op_factory(
                 op_class, wrapper_name, len(inputs), call_args.keys())(**init_args)
 
         return _stateless_operators_cache[key](inputs, call_args)
 
     return wrapper
 
 
-_wrap_stateless.disqualified_arguments = {
-    'bytes_per_sample_hint',
-    'preserve',
-    'seed'
-}
+def _wrap_stateful(op_class, op_name, wrapper_name):
+    """Wraps stateful Eager Operator as method of a class. Callable the same way as functions in
+    fn API, but directly with TensorLists.
+    """
+
+    def wrapper(self, *inputs, **kwargs):
+        inputs, init_args, call_args = _prep_args(
+            inputs, kwargs, op_name, wrapper_name, _callable_op_factory.disqualified_arguments)
+
+        key = op_name + _desc_call_args(inputs, call_args) + str(sorted(init_args.items()))
+
+        if key not in self._operator_cache:
+            seed = self._seed_generator.integers(2**32)
```

Comment:
This is the tricky part here, can you add a comment why do we create seeds this way and what do we cache (that key - so the scalar arguments and the input dim/type maps to a distinct seed).

Reply:
done